schedule/management/commands/reload_courses.py

```diff
@@ -76,19 +77,39 @@ def parse_time_table(course, table):
             for week in parse_weeks(values[5]):
                 offset = timedelta(days=7*(week-1))
                 try:
+                    prof = values[6].title()
+                    if len(prof.split()) == 2:
+                        last, first = prof.split()
+                        prof = "%s %s" % (first, last)
                     Schedule.objects.get_or_create(
                         course=course,
                         start_time=tz.localize(start_time + offset),
                         end_time=tz.localize(end_time + offset),
                         location=values[7],
-                        professor=values[6])
+                        professor=prof)
                 except:
                     course.err_count += 1
 
 
 def parse_course_header(table):
+    title = table.select("span.label-1-0-0")[0].text.strip().replace('\n', ' ')
+    match = re.match(r'(.*)[ ]+(\d+)[ ]+(credits|credtis)[ ]+(\d+\w+)', title)
+    if match:
+        name, credits, _, course_id = match.groups()
+    else:
+        match = re.match(r'(.*)[ ]+(\d+)[ ]+(credits|credtis)', title)
+        if match:
+            name, credits, _ = match.groups()
+            course_id = ""
+        else:
+            name = title
+            credits = -1
+            course_id = ""
+
     course, created = Course.objects.get_or_create(
-        name=table.select("span.label-1-0-0")[0].text.strip(),
+        name=name,
+        credits=int(credits),
+        vub_id=course_id,
```

Comment:
Not sure how `course_id` could have a value different from `""`

Reply:
From 
```
match = re.match(r'(.*)[ ]+(\d+)[ ]+(credits|credtis)[ ]+(\d+\w+)', title)
name, credits, _, course_id = match.groups()
```
(Formats look like this since this year: `Adult Stem and Progenitor Cells 6 credits 4009572ENR`)

Reply:
Getting tired here 😫